assets/prototype/application/ui/components/layout/eeditorModal/useEditorModalManager.ts

```diff
@@ -0,0 +1,80 @@
+import { useReducer, useEffect } from 'react';
+
+import {
+	EditorState,
+	EditorId,
+	EditorStateManager,
+	EditorStateReducer,
+	ActionType,
+	OpenEditorOptions,
+	EntityId,
+} from './types';
+
+export const INITIAL_STATE: EditorState = {
+	isOpen: false,
+	editorId: null,
+	entityId: '',
+};
+
+const useEditorModalManager = (): EditorStateManager => {
+	const [state, dispatch] = useReducer<EditorStateReducer>(reducer, INITIAL_STATE);
+
+	useEffect(() => {
+		console.log('<<<EditorModalManager>>>', state);
+	}, [state]);
+
+	const openEditor = (options: OpenEditorOptions): void => {
+		dispatch({
+			...options,
+			type: ActionType.OPEN_EDITOR,
+			isOpen: true,
+		});
+	};
+
+	const setEditorId = (editorId: EditorId): void => {
+		dispatch({
+			type: ActionType.SET_EDITOR_ID,
+			editorId,
+		});
+	};
+
+	const setEntityId = (entityId: EntityId): void => {
+		dispatch({
+			type: ActionType.SET_ENTITY_ID,
+			entityId,
+		});
+	};
+
+	const closeEditor = (): void => {
+		dispatch({
+			type: ActionType.CLOSE_EDITOR,
+		});
+	};
+
+	return {
+		...state,
+		setEditorId,
+		setEntityId,
+		closeEditor,
+		openEditor,
+	};
+};
+
+const reducer: EditorStateReducer = (state, action) => {
+	const { type, ...rest } = action;
+	const { editorId, entityId } = rest;
+	switch (type) {
+		case ActionType.OPEN_EDITOR:
+			return { ...state, ...rest };
+		case ActionType.SET_EDITOR_ID:
+			return { ...state, editorId };
+		case ActionType.SET_ENTITY_ID:
+			return { ...state, entityId };
+		case ActionType.CLOSE_EDITOR:
+			return INITIAL_STATE;
+		default:
+			throw new Error('Unexpected action');
+	}
+};
```

Comment:
That will add to the complications because we render single instance of `<EditorModal />` instead of rendering multiple instances and adding them to the stack. I will try to find a way to solve that problem.

Reply:
One more issue is that if we create a modal stack and use the same common modal for all, every previous modal would loose its state.